### PR TITLE
feat: PRO-976 creation of dynamodb migration manager tool

### PR DESCRIPTION
**Pull request by brandon-kyle-bailey**

### CODE REVIEW

**Review by liamlevesqueP8 — APPROVED** [review]

File: src/core/migrator.ts [hunk 1]
```diff
@@ -70,6 +71,32 @@ export class Migrator extends Umzug implements Generator {
...
 
+  private async _handleTableCreation(client: any, params: any) {
+    const tablesResult = await client.listTables().promise();
+    const tables = tablesResult.TableNames;
+    if (tables.length > 0 && !tables.includes(this.tableName)) {
+      await client.createTable(params).promise();
+    }
+  }
+
+  private async _awaitTableActiveStatus(client: any, retries = 5) {
```
Comment (gaffleckp8) [inline src/core/migrator.ts:82]: this feels a bit close to me, maybe give it more time than 5 seconds by default?

**Review by gaffleckp8 — APPROVED** [review]

File: src/core/migrator.ts (hunk 1 above)
Comment (brandon-kyle-bailey) [reply src/core/migrator.ts:82]: sure thing, i tested with 10 initially but found it was always active after 5 seconds. but will up the threshold

File: src/core/yargs.ts [hunk 2]
```diff
@@ -46,13 +47,18 @@ export function baseOptions(yargs: Argv<BaseCliOptions>) {
       describe: 'The DynamoDB primaryKey attribute name',
       default: 'name',
       type: 'string'
+    })
+    .option('typescript', {
+      describe: 'The DynamoDB primaryKey attribute name',
```
Comment (gaffleckp8) [inline src/core/yargs.ts:52]: is this description right?

**Review by gaffleckp8 — APPROVED** [review]

File: src/core/yargs.ts (hunk 2 above)
Comment (brandon-kyle-bailey) [reply src/core/yargs.ts:52]: Lol no, lazy copypasta. will update now

**Pull request merged**